moesifasgi/middleware.py

```diff
@@ -167,39 +167,43 @@ async def dispatch(self, request, call_next):
         if self.DEBUG:
             print("event response time: ", response_time)
 
-        skip = await self.logger_helper.should_skip(self.moesif_settings, request, response, self.DEBUG)
-        if not skip:
-            random_percentage = random.random() * 100
-
-            self.sampling_percentage = self.app_config.get_sampling_percentage(self.config,
-                                                                               await self.logger_helper.get_user_id(self.moesif_settings, request, response, request.headers, self.DEBUG),
-                                                                               await self.logger_helper.get_company_id(self.moesif_settings, request, response, self.DEBUG))
-
-            if self.sampling_percentage >= random_percentage:
-                # Prepare Event Request Model
-                event_req = self.event_mapper.to_request(request, request_time, request_body, self.api_version, self.disable_transaction_id)
-
-                # Read Response Body
-                resp_body = None
-                if self.LOG_BODY:
-                    # Consuming FastAPI response and grabbing body here
-                    resp_body = [section async for section in response.__dict__['body_iterator']]
-                    # Preparing FastAPI response
-                    response.__setattr__('body_iterator', async_iterator_wrapper(resp_body))
-
-                # Prepare Event Response Model
-                event_rsp = self.event_mapper.to_response(response, response_time, resp_body)
-                # Prepare Event Model
-                event_data = await self.event_mapper.to_event(request, response, event_req, event_rsp, self.moesif_settings,
-                                                         self.DEBUG)
-
-                # Mask Event Model
-                if self.logger_helper.is_coroutine_function(self.logger_helper.mask_event):
-                    event_data = await self.logger_helper.mask_event(event_data, self.moesif_settings, self.DEBUG)
-                else:
-                    event_data = self.logger_helper.mask_event(event_data, self.moesif_settings, self.DEBUG)
+            # Prepare Event Request Model
+        event_req = self.event_mapper.to_request(request, request_time, request_body, self.api_version,
```

Comment:
why this part is moved outside of the sampling rate? In new change, we are constructing few items but throw it away if sampling rate condition is not met.

Reply:
Since the `get_sampling_rate` function in `moesifpythonrequest` needs the `event_data` process and pass in as a param

Reply:
@XinyuLuan still it should be done only if skip is false.